src/sparse_blas/common_op_verification.hpp

```diff
@@ -0,0 +1,142 @@
+    }
+
+    if (A_view.diag_view != oneapi::mkl::diag::nonunit) {
+        throw mkl::invalid_argument("sparse_blas", function_name,
+                                    "Matrix's diag_view must be `nonunit`.");
+    }
+}
+
+template <typename InternalSparseMatHandleT>
+void check_valid_spmv_common(const std::string &function_name, oneapi::mkl::transpose /*opA*/,
+                             oneapi::mkl::sparse::matrix_view A_view,
+                             InternalSparseMatHandleT internal_A_handle,
+                             oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+                             oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+                             bool is_alpha_host_accessible, bool is_beta_host_accessible) {
+    THROW_IF_NULLPTR(function_name, internal_A_handle);
+    THROW_IF_NULLPTR(function_name, x_handle);
+    THROW_IF_NULLPTR(function_name, y_handle);
+
+    check_all_containers_compatible(function_name, internal_A_handle, x_handle, y_handle);
+    if (internal_A_handle->all_use_buffer()) {
+        check_ptr_is_host_accessible("spmv", "alpha", is_alpha_host_accessible);
+        check_ptr_is_host_accessible("spmv", "beta", is_beta_host_accessible);
+    }
+    if (is_alpha_host_accessible != is_beta_host_accessible) {
+        throw mkl::invalid_argument(
+            "sparse_blas", function_name,
+            "Alpha and beta must both be placed on host memory or device memory.");
+    }
+    if (A_view.type_view == oneapi::mkl::sparse::matrix_descr::diagonal) {
+        throw mkl::invalid_argument("sparse_blas", function_name,
+                                    "Matrix view's type cannot be diagonal.");
+    }
+
+    if (A_view.type_view != oneapi::mkl::sparse::matrix_descr::triangular &&
+        A_view.diag_view == oneapi::mkl::diag::unit) {
+        throw mkl::invalid_argument(
+            "sparse_blas", function_name,
+            "`unit` diag_view can only be used with a triangular type_view.");
+    }
+}
+
+template <typename InternalSparseMatHandleT>
+void check_valid_spsv_common(const std::string &function_name,
+                             oneapi::mkl::sparse::matrix_view A_view,
+                             InternalSparseMatHandleT internal_A_handle,
+                             oneapi::mkl::sparse::dense_vector_handle_t x_handle,
+                             oneapi::mkl::sparse::dense_vector_handle_t y_handle,
+                             bool is_alpha_host_accessible) {
+    THROW_IF_NULLPTR(function_name, internal_A_handle);
+    THROW_IF_NULLPTR(function_name, x_handle);
+    THROW_IF_NULLPTR(function_name, y_handle);
+
+    check_all_containers_compatible(function_name, internal_A_handle, x_handle, y_handle);
+    if (A_view.type_view != matrix_descr::triangular) {
+        throw mkl::invalid_argument("sparse_blas", function_name,
+                                    "Matrix view's type must be `matrix_descr::triangular`.");
```

Comment:
```suggestion
                                    "Matrix view's `type_view` must be `matrix_descr::triangular`.");
```

Reply:
Reworded in https://github.com/oneapi-src/oneMKL/pull/527/commits/7aa5177635e730a6f394783d996912de40a7be7c